src/main/java/org/springframework/data/couchbase/core/ReactiveInsertByIdOperationSupport.java

```diff
@@ -54,28 +54,28 @@ public ReactiveInsertByIdOperationSupport(final ReactiveCouchbaseTemplate templa
 		private final ReplicateTo replicateTo;
 		private final DurabilityLevel durabilityLevel;
 		private final Duration expiry;
+		private final ReactiveTemplateSupport support;
 
 		ReactiveInsertByIdSupport(final ReactiveCouchbaseTemplate template, final Class<T> domainType,
 				final String collection, final PersistTo persistTo, final ReplicateTo replicateTo,
-				final DurabilityLevel durabilityLevel, Duration expiry) {
+				final DurabilityLevel durabilityLevel, Duration expiry, ReactiveTemplateSupport support) {
 			this.template = template;
 			this.domainType = domainType;
 			this.collection = collection;
 			this.persistTo = persistTo;
 			this.replicateTo = replicateTo;
 			this.durabilityLevel = durabilityLevel;
 			this.expiry = expiry;
+			this.support = support;
 		}
 
 		@Override
 		public Mono<T> one(T object) {
-			return Mono.just(object).flatMap(o -> {
-				CouchbaseDocument converted = template.support().encodeEntity(o);
+			return (Mono<T>) Mono.just(object).flatMap(support::encodeEntity).flatMap(converted -> {
 				return template.getCollection(collection).reactive()
-						.insert(converted.getId(), converted.export(), buildInsertOptions(converted)).map(result -> {
-							Object updatedObject = template.support().applyUpdatedId(o, converted.getId());
-							return (T) template.support().applyUpdatedCas(updatedObject, result.cas());
-						});
+						.insert(converted.getId(), converted.export(), buildInsertOptions(converted)).flatMap(result ->
+							support.applyUpdatedId(object, converted.getId())
+								.map(updatedObject -> support.applyUpdatedCas(updatedObject, result.cas())));
 			}).onErrorMap(throwable -> {
```

Comment:
I had test failures here with.  src/test/resources/integration.properties, cluster.type=unmanaged 
mvn integration-test
I think this code needs to be as follows (similar to ReactiveUpsertByIdOperationSupport.one()

```
public Mono<T> one(T object) {
	return (Mono<T>)Mono.just(object).flatMap(template.support()::encodeEntity).flatMap(converted ->
			template.getCollection(collection).reactive()
				.insert(converted.getId(), converted.export(), buildInsertOptions(converted)).flatMap(result ->
					template.support().applyUpdatedId(object, converted.getId())
							.flatMap( updatedObject -> template.support().applyUpdatedCas(updatedObject, result.cas())))
	).onErrorMap(throwable -> {
		if (throwable instanceof RuntimeException) {
			return template.potentiallyConvertRuntimeException((RuntimeException) throwable);
		} else {
			return throwable;
		}
	});
}
```

Reply:
Hi @mikereiche 

This test fails because if you enable auditing, field annoted with @CreatedBy, @CreatedDate, @LastModifiedBy and @LastModifiedDate are injected.

IMHO, the test is not entirely correct as the assertion should be between the User obtained from `userRepository.save(user).block()` and the User obtained from userRepository.findById(user.getId()).blockOptional().

Reply:
Please set src/test/resources/integration.properties, cluster.type=unmanaged
and run mvn integration-test.  It gives ClassCastExceptions.
Yes, it should be comparing the entities from save and find. 

[ERROR] Errors: 
[ERROR]   CouchbaseTemplateKeyValueIntegrationTests.existsById:251 ClassCast reactor.cor...
[ERROR]   CouchbaseTemplateKeyValueIntegrationTests.insertById:231 ClassCast reactor.cor...
[ERROR]   CouchbaseTemplateKeyValueIntegrationTests.insertByIdwithDurability:240 ClassCast
[ERROR]   CouchbaseTemplateKeyValueIntegrationTests.saveAndFindImmutableById:266 ClassCast
[ERROR]   CouchbaseTemplateKeyValueIntegrationTests.withDurability:115 ClassCast reactor...
[ERROR]   CouchbaseTemplateKeyValueIntegrationTests.withExpiryAndExpiryAnnotation:157 ClassCast

Reply:
Fixed